harness/determined/common/experimental/determined.py

```diff
@@ -72,10 +72,36 @@ def _from_session(cls, session: api.Session) -> "Determined":
     def create_user(
         self, username: str, admin: bool, password: Optional[str] = None, remote: bool = False
     ) -> user.User:
+        """Creates a user.
+
+        The user's credentials may be managed by a remote service, in which case the `remote`
+        argument should be set to `true`, and then SSO should be configured for the user.
+        Otherwise, a password must be set that meets complexity requirements.
+
+        The complexity requirements are:
+            - Must be at least 8 characters long.
+            - Must contain at least one upper-case letter.
+            - Must contain at least one lower-case letter.
+            - Must contain at least one number.
+
+        Arg:
+            username: username of the user.
+            admin: indicates whether the user is an admin.
+            password: password of the user.
+            remote: indicates whether the user is managed by a remote service.
```

Comment:
If you think it helpful for a user (I don't have a good conception of what they look like so I don't know if it'd be obvious to them or not), this would be a good place to record that `remote` means "doesn't have a password, cannot log in with one." Maybe also a place to note that this only works if EE is enabled? Not sure how to handle that...

You and I both know these facts. Not sure a user necessarily will (genuinely not sure)

Reply:
Nice. This is a superb docstring.